### DISCUSSION TITLE
Importing functions from other custom script?

### DISCUSSION THREAD

**Discussion by PaulWestphal** [netbox-community/netbox#7735] [Q&A] [answered]
Hello. We use a lot of custom scripts in our netbox instance, and we are at a point where we need to reuse code.
For this we created a `utils.py` in our scripts folder and tried to import it using `import utils`. This leads to a `module not found` error.

Is there a way to import plain python files without building and installing as a package?

**Comment by jeremystretch** [comment, marked as answer]
There's nothing special about importing other Python modules from within a script; they just need to be accessible from within the Python environment. Are you sure it's within the same directory as the module importing it? Maybe try `from . import utils`.

**Reply by kkthxbye-code** [reply]
I think you need `from scripts import utils` - at least I do both in my docker production setup and my development setup (and the SCRIPT_HOME variable has to be the default value)

Plain import doesn't work because the script folder is not in the python path. from . import utils doesn't work because of `attempted relative import with no known parent package`.

**Reply by PaulWestphal** [reply]
@kkthxbye-code This suggestion worked, thanks so much!
